tests/CounterInterceptor/MyWorkflowTests.cs

```diff
@@ -0,0 +1,76 @@
+using Temporalio.Client;
+using Temporalio.Testing;
+using Temporalio.Worker;
+using TemporalioSamples.CounterInterceptor;
+using Xunit;
+
+namespace TemporalioSamples.Tests.CounterInterceptor;
+
+public class MyWorkflowTests
+{
+    [Fact]
+    public async Task RunAsync_CounterInterceptor()
+    {
+        await using var env = await WorkflowEnvironment.StartLocalAsync();
+
+        // add the interceptor to the client
+        var clientOptions = (TemporalClientOptions)env.Client.Options.Clone();
+        clientOptions.Interceptors = new[]
+        {
+            new SimpleClientCallsInterceptor(),
+        };
+
+        var client = new TemporalClient(env.Client.Connection, clientOptions);
+
+        var workerOptions = new TemporalWorkerOptions(
+            TemporalioSamples.CounterInterceptor.Constants.TaskQueue).
+                AddAllActivities(new MyActivities()).
+                AddWorkflow<MyWorkflow>().
+                AddWorkflow<MyChildWorkflow>();
+
+        workerOptions.Interceptors = new[] { new SimpleCounterWorkerInterceptor() };
+
+        var parentWorkflowId = "ParentWorkflowId";
+        var childWorkflowId = TemporalioSamples.CounterInterceptor.Constants.ChildWorkflowId;
+
+        using var worker = new TemporalWorker(client, workerOptions);
+        await worker.ExecuteAsync(async () =>
+        {
+            var handle = await client.StartWorkflowAsync(
+                (MyWorkflow wf) => wf.ExecAsync(),
+                new(
+                    id: parentWorkflowId,
+                    taskQueue: TemporalioSamples.CounterInterceptor.Constants.TaskQueue));
+
+            await handle.SignalAsync(wf => wf.SignalNameAndTitleAsync("John", "Customer"));
+
+            var name = await handle.QueryAsync(wf => wf.QueryName());
+            var title = await handle.QueryAsync(wf => wf.QueryTitle());
+
+            // send exit signal to workflow
+            await handle.SignalAsync(wf => wf.ExitAsync());
+
+            // Wait for the workflow to complete
+            var result = await handle.GetResultAsync();
+
+            // validate that the worker counters have the correct numbers for the parent
```

Comment:
nit: I have an allergy to comments that start with lower case. Casing and grammar reflect professionalism and attention to detail, and enhance readability, which are critical in a professional environment.

Reply:
Fixed